### ISSUE TITLE
Quadratic programs with CPLEX and Gurobi

### ISSUE THREAD

**Issue by dafrick** [JuliaOpt/MathProgBase.jl#52]
I have an issue when using quadprog with both CPLEX and Gurobi, this seems to be due to the fact that they don't support range constraints, i.e. either rowlb or rowub has to be Inf. This would be fine but unfortunately, if I use senses and my problem data are integers then the one-sided constraints are converted to the smallest (or biggest) Int64, not Inf which makes both CPLEX als well as Gurobi trip up.
This should be fixed in quadprog.jl:29 and 33, by replacing typemin(realtype) and typemax(realtype) with -Inf and Inf respectively.
The individual solvers should have to decide how they want to deal with Inf.

**Comment by joehuchette** [comment]
The problem with this fix is that there is no notion of Inf for Int64, so trying to do an assignment like 
```julia
rowub = Array(Int64, 1)
rowub[1] = Inf # => ERROR: InexactError()
```
will fail. I'm not sure what the best fix for the general problem here is TBH, though converting your data to Float64 should fix your immediate issue. cc @mlubin @IainNZ.

(I should note that at least CPLEX supports QPs with ranged constraints [I'm solving some right now], but that duals aren't supported in an intelligent way, hence the warning message).

**Comment by joehuchette** [comment]
Also: why are you using the ``quadprog`` interface, as opposed to a modeling layer like JuMP or Convex? As you might have guessed, these high level interfaces haven't gotten much love, largely because we haven't hit use-cases where they're to be preferred over using an algebraic modeling tool.

**Comment by dafrick** [comment]
I agree that converting to float is an immediate fix and I have already done that, I just would like to spare someone finding this out the hard way.
The problem is that if you don't take care of converting to float then CPLEX will still run, but in my case the result was simply wrong, I'm not sure why exactly, could be due to numerical issues.
I suspected that the issue cannot be resolved in a straightforward way and maybe it is up to CPLEX.jl and Gurobi.jl to deal with it, but conceptually you replace a single-sided constraint with a two-sided constraint where one constraint is really far away. It is obvious that this is not the same problem and apparently this leads to issues with some of the solvers. 

Since this problem is not an exotic one and can happen easily - the problem I was solving had 4 variables and 3-4 constraints, I would urge you to at least add a warning with specific instructions on how to avoid the issue in that case.

Sidenote: I prefer using quadprog, because I prefer knowing the exact problem I'm solving and for small but parametric problems I'm faster setting up the problem directly rather than reading the documentation of JuMP and modelling it that way.

**Comment by joehuchette** [comment]
You're certainly right, we can at least add a warning until we figure out how best to handle something like this. If it's not too much trouble, we'd love a minimal example of your formulation that produced an incorrect result.

**Comment by dafrick** [comment]
I was looking for a minimal example, and I found the real problem:
```julia
sol1 = quadprog(zeros(4), eye(4,4), [0 0 0 0; 0 0 1 0; 0 0 0 1], ['<', char('='*ones(2))], [1, 0, 2], -Inf*ones(4), Inf*ones(4), CplexSolver())
sol2 =  quadprog(zeros(4), eye(4,4), [0 0 0 0; 0 0 1 0; 0 0 0 1], ['<', '='*ones(2)], [1, 0, 2], -Inf*ones(4), Inf*ones(4), CplexSolver())
````
The parameter `rowlb` either takes the lower bound constraints or the `senses`, I made the mistake of implicitly converting the characters to numbers which were then interpreted as constraints. When this apparent error was fixed by writing the `rowlb` explicitly I assumed the problem was with the conversion from senses to `rowlb`.

This is an error that is easy to make, but I'm not sure it warrants a warning.
